drud/ddev
# Reorganize web and db containers to have real user and hostname, persistent bash history, for #926

## The Problem/Issue/Bug:

This reorganizes the web and db containers to have a real named user which gets added at docker build time. It is intended to be the same username and id as on the host, and gets added dynamically.

It also sets the hostname inside the container to a useful value, like `<projectname>-web`

This sets the stage for several related pieces, including #926 (user .bashrc additions)

## TODO
- [x] Make sure that oddball uids have a fallback
- [x] Docs? In some ways this is just a better experience and doesn't need more docs.
- [x] Make bgsync container work with the same strategy.

## Manual Testing Instructions:

* `ddev start`
* `ddev ssh` and `ddev ssh -s db` - you should see a real user created with your username, and the hostname should be a nice hostname. There should be actual files in ~/, and the homedir is no longer /home.
* `ddev restart` and `ddev ssh` and you should be able to up/down through bash history from previous session. `cat $HISTFILE` should show you all that.

## Automated Testing Overview:

## Related Issue Link(s):

#926 is for lots of related things, includes the .bash_history, but this doesn't solve all yet, it's a prereq.

## Release/Deployment notes:
